plugins/woocommerce/templates/block-notices/error.php

```diff
@@ -27,7 +27,7 @@
 
 ?>
 
-<div class="wc-block-components-notice-banner is-error" role="alert" <?php echo $multiple ? '' : wc_get_notice_data_attr( $notices[0] ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
```

Comment:
>Why are we removing the role="alert" attribute here? Most (or all?) of the block-notices templates have that, so I guess there must be a reason or at least we should revaluate for consistency.

That's a good point. For some reason, the other day while I was testing, having role="alert" didn't allow the screen reader to read the multi-line messages, but when I tested again on Safari (and Chrome) today, it was working fine, so I believe it depends on the browser as well.

Another important thing is that `role="alert"` is not supposed to be used in a full page reload scenario.
![image](https://github.com/woocommerce/woocommerce/assets/15019298/c3bb65c7-a6b2-47dc-a26b-74d1f850b8c5)


That said, 
- since it is a block template, I am assuming that it will be used in scenarios where we update the message inside the error container dynamically.
- A bunch of e2e tests also depend on this attribute, keeping all the factors in mind, it makes sense to keep this attribute.